filebeat/config/main_test.go

```diff
@@ -0,0 +1,37 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	err := flag.Set("strict.perms", "false")
+	if err != nil {
+		fmt.Fprintln(os.Stderr,
+			fmt.Sprintf("failed to set flag strict.perms=false: %v", err))
+		os.Exit(1)
+	}
+	flag.Parse()
+
+	os.Exit(m.Run())
+}
```

Comment:
Since it's the same code in 3 files it would be better to extract it in a helper function and call that function instead. Otherwise, if we needed to change it in the future we would have to remember about 3 different places.

Reply:
done in https://github.com/elastic/beats/pull/42914/commits/423c54408b2f33937d3a8057061e80c382e7db1c